great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -724,7 +724,8 @@ def delete_datasource(
             raise ValueError(f"Datasource {datasource_name} not found")
 
         if save_changes:
-            self._datasource_store.delete_by_name(datasource_name)  # type: ignore[attr-defined]
+            datasource_config = datasourceConfigSchema.load(datasource.config)
```

Comment:
Sorry, not part of this PR, but for line `718` should that conditional be tweaked to account for empty strings as well.
```python
        if not datasource_name:
            raise ValueError("Datasource names must be a datasource name")
```

Reply:
an easy fix! good catch, I've changed this in the next commit